framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -747,4 +753,60 @@ public static String removeNewLines(String input) {
     return builder.toString();
   }
 
+  /* Get DrillTestCase for a single query file */
+  public static void singleTestCase(File singleTestFile,
+                                      String[] testGroups,
+                                      List<DrillTestCase> drillTestCases)
+      throws IOException {
+    List<File> jsonFiles = searchFiles(singleTestFile.getParentFile(), ".*.json");
+    if (jsonFiles.isEmpty()) {
+      String updir = "";
+      while (jsonFiles.isEmpty()) {
```

Comment:
What happens if you never encounter a test definition file in any of the upstream directories?